opencti-platform/opencti-graphql/src/utils/stix-filtering/stix-testers.ts

```diff
@@ -0,0 +1,377 @@
+ */
+export const testRefs = (stix: any, filter: Filter) => {
+  const stixValues: string[] = stixRefsExtractor(stix, generateStandardId);
+  return testStringFilter(filter, stixValues);
+};
+
+/**
+ * INSTANCE (elementId)
+ * - elementId is id in stix extension
+ */
+export const testInstance = (stix: any, filter: Filter) => {
+  const stixValues: string[] = toValidArray(stix.extensions?.[STIX_EXT_OCTI]?.id);
+  return testStringFilter(filter, stixValues);
+};
+
+/**
+ * CONNECTED TO
+ - depending on stix type (relation or sighting), we might search in different paths, aggregated
+ */
+export const testConnectedTo = (stix: any, filter: Filter) => {
+  // only applies with "eq" operator
+  if (filter.operator !== 'eq') {
+    return false;
+  }
+
+  // advanced test between filter ids and the entity relations and refs
+  // we shall aggregate all candidate fields and match the filter
+  const aggregatedStixValues = [];
+  if (stix.type === STIX_TYPE_RELATION) {
+    aggregatedStixValues.push(...toValidArray(stix.target_ref)); // to
+    aggregatedStixValues.push(...toValidArray(stix.source_ref)); // from
+  }
+  if (stix.type === STIX_TYPE_SIGHTING) {
+    aggregatedStixValues.push(...(stix.where_sighted_refs ?? [])); // to
+    aggregatedStixValues.push(...toValidArray(stix.sighting_of_ref)); // from
+  }
+  // refs
+  aggregatedStixValues.push(...stixRefsExtractor(stix, generateStandardId));
+
+  return testStringFilter(filter, aggregatedStixValues);
+};
+
+/**
+ * TODO: This mapping could be given by the schema, like we do with stix converters
+ */
+export const FILTER_KEY_TESTERS_MAP: Record<string, TesterFunction> = {
+  // basic keys
+  [ASSIGNEE_FILTER]: testAssignee,
+  [CONFIDENCE_FILTER]: testConfidence,
+  [CREATED_BY_FILTER]: testCreatedBy,
+  [CREATOR_FILTER]: testCreator,
+  [DETECTION_FILTER]: testDetection,
+  [INDICATOR_FILTER]: testIndicator,
+  [LABEL_FILTER]: testLabel,
+  [MAIN_OBSERVABLE_TYPE_FILTER]: testMainObservableType,
+  [MARKING_FILTER]: testMarkingFilter,
+  [OBJECT_CONTAINS_FILTER]: testObjectContains,
+  [PATTERN_FILTER]: testPattern,
+  [PRIORITY_FILTER]: testPriority,
+  [REVOKED_FILTER]: testRevoked,
+  [SEVERITY_FILTER]: testSeverity,
+  [SCORE_FILTER]: testScore,
+  [TYPE_FILTER]: testEntityType,
+  [WORKFLOW_FILTER]: testWorkflow,
+  [INSTANCE_FILTER]: testInstance,
+
+  // special keys (more complex behavior)
+  [CONECTED_TO_FILTER]: testConnectedTo,
+  [RELATION_FROM]: testRelationFrom,
+  [RELATION_FROM_TYPES]: testRelationFromTypes,
+  [RELATION_TO]: testRelationTo,
+  [RELATION_TO_TYPES]: testRelationToTypes,
+};
+
+/**
+ * Gives the right tester function according to the filter key.
+ * If the key is not handled, returns a function that always return false.
+ * @param key
+ */
+export const getStixTesterFromFilterKey = (key: string): TesterFunction => {
+  if (!FILTER_KEY_TESTERS_MAP[key]) {
+    logApp.warn(`Unrecognized filter key when matching stix object: [${key}]`);
+    return () => false;
+  }
```

Comment:
better to validate filter keys (no multiple + exist in map) and throw error on top before matching.
Warn is too lazy.